src/sessions/PageManager.ts

```diff
@@ -17,11 +17,19 @@ export type Attributes = {
     parentPageId?: string;
     interaction?: number;
     pageTags?: string[];
+    // The value types of custom attributes are restricted to the types: string | number | boolean
+    // However, given that pageTags is a string array, we need to include it as a valid type
+    // Events will be verified by our service to validate attribute value types where
+    //  1) pageTags attribute value must be of type string[]
+    //  2) any other attribute value must be of the following types:
+    //      string | number | boolean
+    [k: string]: string | number | boolean | string[];
 };
 
 export type PageAttributes = {
     pageId: string;
     pageTags?: string[];
+    [k: string]: string | number | boolean | string[];
```

Comment:
nit: Additional attributes can only be primitives, right?
```suggestion
    [k: string]: string | number | boolean;
```

Reply:
As mentioned above, I think we need to still allow values to have `string[]` type to avoid the error
`Property 'pageTags' of type 'string[]' is not assignable to 'string' index type 'string | number | boolean'.` 
Metadata schema will validate attributes where we drop any attributes with `string[]` type if it's not the `pageTags` attribute.